Mathlib/Data/Seq/Defs.lean

```diff
@@ -377,6 +378,37 @@ theorem eq_of_bisim (bisim : IsBisimulation R) {s₁ s₂} (r : s₁ ~ s₂) : s
       · simp
   · exact ⟨s₁, s₂, rfl, rfl, r⟩
 
+/-- Version of `eq_of_bisim` that looks more like an induction principle. -/
+theorem eq_of_bisim' {s₁ s₂ : Seq α}
+    (motive : Seq α → Seq α → Prop)
+    (base : motive s₁ s₂)
+    (step : ∀ s₁ s₂, motive s₁ s₂ →
+      (∃ x s₁' s₂', s₁ = cons x s₁' ∧ s₂ = cons x s₂' ∧ motive s₁' s₂') ∨
+      (s₁ = nil ∧ s₂ = nil)) : s₁ = s₂ := by
+  apply eq_of_bisim motive _ base
+  intro s₁ s₂ h
+  rcases step s₁ s₂ h with (⟨_, _, _, h₁, h₂, _⟩ | ⟨h_nil₁, h_nil₂⟩)
+  · simpa [h₁, h₂]
+  · simp [h_nil₁, h_nil₂]
+
+/-- Version of `eq_of_bisim'` that requires only `s₁ = s₂`
+instead of `s₁ = nil ∧ s₂ = nil` in `step`. -/
+theorem eq_of_bisim_strong {s₁ s₂ : Seq α}
+    (motive : Seq α → Seq α → Prop)
+    (base : motive s₁ s₂)
+    (step : ∀ s₁ s₂, motive s₁ s₂ →
+      (s₁ = s₂) ∨
```

Comment:
Is there any reason why the order of clauses here is swapped compared to `eq_of_bisim'`?